src/IECoreMaya/SceneShapeInterface.cpp

```diff
@@ -824,6 +825,34 @@ MStatus SceneShapeInterface::compute( const MPlug &plug, MDataBlock &dataBlock )
 					{
 						MFnNurbsCurveData fnData;
 						data = fnData.create();
+
+						// If "index" convert paramter is specified, it determines which one of the curves to output in the Curves primitive.
+						std::map<std::string, std::string> convertParams;
+						bool isParamsRead = readConvertParams( convertParams, index );
```

Comment:
I don't know what the use case is for all this stuff, but if the parameter values aren't entered by users directly, one other option is to actually store IECore::Objects directly on a plug, so no conversion would be necessary at all.